www/js/control/ProfileSettings.jsx

```diff
@@ -0,0 +1,286 @@
+               <SettingRow textKey="control.check-sensed-data" iconName="arrow-expand-right" action={showSensed}></SettingRow>
+               <SettingRow textKey="control.collection" iconName="pencil" action={editCollectionConfig}></SettingRow>
+               <ControlDataTable controlData={settings?.collect?.show_config}></ControlDataTable>
+               <SettingRow textKey="control.sync" iconName="pencil" action={editSyncConfig}></SettingRow>
+               <ControlDataTable controlData={settings?.sync?.show_config}></ControlDataTable>
+               <SettingRow textKey="control.app-version" iconName="application" action={()=>console.log("")} desc={settings?.clientAppVer}></SettingRow>
+           </ExpansionSection>
+
+        {/* menu for "nuke data" */}
+            <Modal visible={nukeSetVis} onDismiss={() => setNukeVis(false)}
+            transparent={true}>
+                <Dialog visible={nukeSetVis}
+                onDismiss={() => setNukeVis(false)}>
+                    <Dialog.Title>{t('general-settings.clear-data')}</Dialog.Title>
+                    <Dialog.Content>
+                        <Button onPress={() => {KVStore.clearOnlyLocal;
+                                                setNukeVis(false);}}>
+                            {t('general-settings.nuke-ui-state-only')}
+                        </Button>
+                        <Button onPress={() => {KVStore.clearOnlyNative;
+                                                setNukeVis(false);}}>
+                            {t('general-settings.nuke-native-cache-only')}
+                        </Button>
+                        <Button onPress={() => {KVStore.clearAll;
+                                                setNukeVis(false);}}>
+                            {t('general-settings.nuke-everything')}
+                        </Button>
+                    </Dialog.Content>
+                    <Dialog.Actions>
+                        <Button onPress={() => setNukeVis(false)}>{t('general-settings.cancel')}</Button>
+                    </Dialog.Actions>
+                </Dialog>
+            </Modal>
+
+        {/* menu for "set carbon dataset - only somewhat working" */}
+            <Modal visible={carbonDataVis} onDismiss={() => setCarbonDataVis(false)}
+                transparent={true}>
+                <Dialog visible={carbonDataVis}
+                    onDismiss={() => setCarbonDataVis(false)}>
+                    <Dialog.Title>{t('general-settings.choose-dataset')}</Dialog.Title>
+                    <Dialog.Content>
+                        {carbonOptions.map((e) =>
+                            <Button key={e.text}
+                            onPress={() =>  {
+                                console.log("changeCarbonDataset(): chose locale " + e.value);
+                                CarbonDatasetHelper.saveCurrentCarbonDatasetLocale(e.value); //there's some sort of error here
```

Comment:
There is an error being thrown on this line, and I actually found the same error being thrown in the same place on master_for_platform. Should I try to change CarbonDatasetHelper to resolve the issue? I can see how changing this value is meant to change the amount of emissions emitted by each mode, is this feature still important?
![Screenshot 2023-07-06 at 2 14 53 PM](https://github.com/e-mission/e-mission-phone/assets/54848919/b468eaca-8ddc-4999-8f9b-ca9ac6f5a1e4)

Reply:
If it is not a regression, I am fine with moving this to production. It is not super important, and we can fix it while fixing the dashboard.

We should also file a more generic issue for it, because arguably we want to automatically figure out the CO2 emissions based on the location of the trip, instead of having the user choose it. The grid in WA state has a lot of hydropower, for example, while the grid in the Southern states uses more coal.

Note also that this is not just location dependent, it is also time dependent. We are currently using values from 2020, so are not taking into account the impact of efforts to decarbonize the grid by using wind and solar.

EIA API integration and caching, @nataliejschultz?

Reply:
I opened an issue to track this [#954](https://github.com/e-mission/e-mission-docs/issues/954)